src/main/java/net/fabricmc/tinyremapper/TinyRemapper.java

```diff
@@ -244,7 +244,13 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
 			}
 
 			if (fs == null) {
-				fs = FileSystems.newFileSystem(uri, Collections.emptyMap());
+                Map<String, Object> env = new HashMap<>();
+			    if(System.getProperty("tiny_use_zipcache","FALSE").equals("TRUE")){
```

Comment:
Do people use lowercase or uppercase for values more often?

Reply:
Better yet, use lowercase and call toLowerCase on the property value. Furthermore, System.getProperty is null if the property isn't present, so this will crash with an NPE if the property isn't specified.

Reply:
It can't do that, because I give a default value to be used when it's not present. I can set it to lowercase.

Although, it's only a partial fix, because it still runs out of memory, but it's no longer because of the zipfs.

Reply:
My suggestion is to use `"tiny.useZipCache"`

Also - don't forget `Boolean.getBoolean()` :wink: